From f8b0126f7f1f0b4093124408d73954963f6a87ba Mon Sep 17 00:00:00 2001
From: Philip Withnall <philip@tecnocode.co.uk>
Date: Wed, 25 Oct 2023 11:20:36 +0100
Subject: [PATCH 21/24] build: Change configure_file() to fs.copyfile() when
 copy is set

Meson has preferred `fs.copyfile()` since 0.62. This fixes a set of
Meson warnings.

Signed-off-by: Philip Withnall <philip@tecnocode.co.uk>
(cherry picked from commit 4ce58df8540f25f823418a1d45ddb07ada52ec22)
---
 docs/reference/gio/meson.build    | 6 +-----
 gio/gdbus-2.0/codegen/meson.build | 8 ++++----
 gio/meson.build                   | 2 ++
 gio/tests/meson.build             | 5 +----
 4 files changed, 8 insertions(+), 13 deletions(-)

diff --git a/docs/reference/gio/meson.build b/docs/reference/gio/meson.build
index 8c8e15fd8..8ebdd9ad6 100644
--- a/docs/reference/gio/meson.build
+++ b/docs/reference/gio/meson.build
@@ -153,11 +153,7 @@ if get_option('gtk_doc')
     command : [concat_files_helper, '@OUTPUT@', '@INPUT@'],
   )
 
-  configure_file(
-    output : 'gio-docs-platform.xml',
-    input : platform_file,
-    copy : true,
-  )
+  fs.copyfile(platform_file, 'gio-docs-platform.xml')
 
   content_files = [
     'overview.xml',
diff --git a/gio/gdbus-2.0/codegen/meson.build b/gio/gdbus-2.0/codegen/meson.build
index 16abd97d6..68e216803 100644
--- a/gio/gdbus-2.0/codegen/meson.build
+++ b/gio/gdbus-2.0/codegen/meson.build
@@ -29,6 +29,7 @@ meson.override_find_program('gdbus-codegen', gdbus_codegen)
 codegen_dir = join_paths(glib_datadir, 'glib-2.0', 'codegen')
 
 gdbus_codegen_built_files = []
+gdbus_codegen_built_targets = []
 gdbus_codegen_built_files += configure_file(input : 'config.py.in',
                                             output : 'config.py',
                                             install_dir : codegen_dir,
@@ -38,8 +39,7 @@ gdbus_codegen_built_files += configure_file(input : 'config.py.in',
 foreach f : gdbus_codegen_files
   # Copy these into the builddir so that gdbus-codegen can be used uninstalled
   # and then install it too so that it can be used after installation
-  gdbus_codegen_built_files += configure_file(input : f, output : f,
-                                              install_dir : codegen_dir,
-                                              install_tag : 'bin-devel',
-                                              copy : true)
+  gdbus_codegen_built_targets += fs.copyfile(f, f,
+                                             install_dir : codegen_dir,
+                                             install_tag : 'bin-devel')
 endforeach
diff --git a/gio/meson.build b/gio/meson.build
index 14f0d8259..d474fa059 100644
--- a/gio/meson.build
+++ b/gio/meson.build
@@ -237,6 +237,7 @@ xdp_dbus_generated = custom_target('xdp-dbus',
              'org.freedesktop.portal.Trash.xml'],
     output : ['xdp-dbus.h', 'xdp-dbus.c'],
     depend_files : gdbus_codegen_built_files,
+    depends : gdbus_codegen_built_targets,
     command : [python, gdbus_codegen,
                '--interface-prefix', 'org.freedesktop.portal.',
                '--output-directory', '@OUTDIR@',
@@ -249,6 +250,7 @@ gdbus_daemon_generated = custom_target('gdbus-daemon-generated',
     input : ['dbus-daemon.xml'],
     output : ['gdbus-daemon-generated.h', 'gdbus-daemon-generated.c'],
     depend_files : gdbus_codegen_built_files,
+    depends : gdbus_codegen_built_targets,
     command : [python, gdbus_codegen,
                '--interface-prefix', 'org.',
                '--output-directory', '@OUTDIR@',
diff --git a/gio/tests/meson.build b/gio/tests/meson.build
index 41d4e3e6c..dbe7c38cc 100644
--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -767,10 +767,7 @@ if meson.can_run_host_binaries()
                '@INPUT@'])
 
   # referenced by test.gresource.xml
-  test_generated_txt = configure_file(input : 'test1.txt',
-    output : 'test-generated.txt',
-    copy : true,
-  )
+  test_generated_txt = fs.copyfile('test1.txt', 'test-generated.txt')
 
   resources_extra_sources = [
     test_gresource,
-- 
2.44.0.windows.1

